go-ds-s3-plugin/main.go

```diff
@@ -0,0 +1,11 @@
+package main
+
+import (
+	plugin "github.com/ipfs/go-ds-s3/src"
+)
+
+var Plugins = plugin.Plugins //nolint
+
+func main() {
```

Comment:
This is suspicious, `s3ds.go` doesn't needed to do this, why does this needs it ?

Reply:
tl;dr: [Line 7](https://github.com/ipfs/go-ds-s3/blob/2c56540bb971423f7f39d46839532f47278fc86d/go-ds-s3-plugin/main.go#L7) is needed, [lines 9-11](https://github.com/ipfs/go-ds-s3/blob/2c56540bb971423f7f39d46839532f47278fc86d/go-ds-s3-plugin/main.go#L9) can be removed. See [`06217e6`](https://github.com/ipfs/go-ds-s3/pull/275/commits/06217e611dfd34889b2dd20479e81e3674783f60)

Thanks for questioning this! I actually copied main.go from [v0.9.0](https://github.com/ipfs/go-ds-s3/tree/v0.9.0) and tried to understand why it was written like this: 

1. Importing the plugin without using it will give GO error. So line 7 was to suppress import error ("nolint"). I tried importing with blank identifier (underscore) as explicit package name. This worked but caused another error:
2. Using plugin unbundled (like .ipfs/plugins/go-ds-s3-plugin) will fail when importing with blank identifier as described above. Kubo (f.E. ipfs version) thows error: `plugin: symbol Plugins not found in plugin`.